frontend/src/components/unionListItem.vue

```diff
@@ -19,18 +19,21 @@
   </a>
 </template>
 
-<script lang="ts">
-// eslint-disable-next-line
-import { PropType, defineComponent } from "vue";
-// eslint-disable-next-line
-import { UnionInterface } from "./unionDashboard.vue";
+<script>
+import router from "@/router";
 
-export default defineComponent({
+export default {
   name: "unionListItem",
   props: {
-    unionfields: Object as PropType<UnionInterface>,
+    unionfields: Object,
   },
-});
+  methods: {
+    onClickUnion() {
+      const name = this.unionfields.union.name;
```

Comment:
Hier zou je nog een if check om heen kunnen zetten of er wel een union bestaat in unionfields.

```
if(this.unionfields.union && this.unionfields.union.name){
    name = this.unionfields.union.name
```

Dan kan je namelijk veilig de value name gaan parsen. (In de else kan je dan bijv naar de home pagina routen.

trouwens `unionfields` is geen goeie variable name. `unionFields`